### PR TITLE
Add Route Creation Map

### PR DESCRIPTION
**Pull request by rbilgil**

### CODE REVIEW

File: packages/route-creation-map/src/utils/background-workers.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
+import FindMatchingShapePointForCoordsWorker from 'worker-loader?inline!../workers/find-matching-shape-point-for-coords-worker.js';
+import MatchStopsWithShapeWorker from 'worker-loader?inline!../workers/match-stops-with-shape-worker.js';
+import GetOverlappingPatternsWorker from 'worker-loader?inline!../workers/get-overlapping-patterns-worker.js';
+import GenerateSpacialIndexWorker from 'worker-loader?inline!../workers/generate-spacial-index-worker.js';
```
Comment (peteris) [inline packages/route-creation-map/src/utils/background-workers.js]: Looks like the spelling correction (`spatial`) got lost in transition here

File: packages/route-creation-map/README.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                    shapePoints={shapePoints}
+                    osrmPointsList={osrmPointsList}
+                    stopsList={stopsList}
+                    onOSRMPointsListUpdated={(newOSRMPointsList) => {
+                        this.setState({ osrmPointsList: newOSRMPointsList }); 
+                    }}
+                    onShapePointsUpdated={(newShapePoints) => {
+                        this.setState({ shapePoints: newShapePoints });
+                    }}
+                    onStopsListUpdated={(newStopsList) => {
```
Comment (peteris) [inline packages/route-creation-map/README.md]: Should we change the names of these methods to `onChangeStopsList`, `onChangeShapePoints`, `onChangeOSRMPointsList`? Or failing that, `onUpdate...`?

File: packages/route-creation-map/README.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+```
+
+## Types
+You can import helpful flow types to annotate your project (more in types/types.js):
+
+```jsx harmony
+// @flow
+import type { StopsList, OSRMPointsList, ShapePoints } from '@citymapper/route-creation-map/types';
+```
```
Comment (peteris) [inline packages/route-creation-map/README.md]: Great to have the README in place right off the bat 🙌

File: packages/route-creation-map/src/route-creation-map.js [hunk 4]
```diff
@@ -0,0 +1,1011 @@
...
+    } = {
+        dragPoint: null,
+        dragId: null,
+        draggingShapePoint: false,
+        currentDraggingPoint: null,
+    };
+
+    dragJustEnded: boolean = false;
+    draggingIntermediatePoint: boolean = false;
+    mousingOverPolyline: boolean = false;
```
Comment (peteris) [inline packages/route-creation-map/src/route-creation-map.js]: Should we take this opportunity to update these names? `dragFinished` and `hoveringPolyline` feels a bit less casual 😄

File: packages/route-creation-map/README.md (hunk 2 above)
Comment (rbilgil) [inline packages/route-creation-map/README.md]: onChange seems good

File: packages/route-creation-map/src/route-creation-map.js (hunk 4 above)
Comment (rbilgil) [inline packages/route-creation-map/src/route-creation-map.js]: 🐭

**Pull request merged**